### DISCUSSION TITLE
New plugin - Segment

### DISCUSSION THREAD

**Discussion by mytototo** [Kong/kong#7388] [Ideas and feature requests]
It would be nice to have a Segment plugin so we can identify and track our customers through a Kong gateway. Also, maybe Kong can be a Segment source.

**Comment by colllin** [comment]
I got a custom `segment-log` plugin working today.  Check it out: https://github.com/DigitalGlobe/kong-segment-log

It's a little tailored to our needs (requires JWT in request Authorization header).  It could be pretty easily adapted to additional situations.  For example, it might be nice if it recognized the consumer_id produced by some Kong authentication plugins.

![image](https://cloud.githubusercontent.com/assets/185923/19948103/9dbf3312-a110-11e6-82a1-1c443e93d2d0.png)
